app/services/xccdf_report_parser.rb

```diff
@@ -25,6 +27,12 @@ def initialize(report_contents, message)
     @metadata = message['metadata']
     @host_inventory_id = message['id']
     @oscap_parser = OpenscapParser::Base.new(report_contents)
+    check_report_format
+  end
+
+  def check_report_format
+    raise WrongFormatError if @oscap_parser.report_xml.
+      at_xpath('.//benchmark/@href').value.match?('.*-xccdf.xml')
```

Comment:
To be safe, we should look only for benchmarks nested under a TestResult. xpath supports matching as well:

```rb
.xpath(".//TestResult/benchmark[contains(@href, '-xccdf.xml')]/@href").any?
```

Reply:
I can do that for sure :smile:  there was only 1 match for this xpath as far as I could see but it's a good point.

Reply:
There _should_ only ever be one `<TestResult />` afaik, but I could be wrong.